src/main/java/net/neoforged/neoforge/fluids/crafting/UnitAmountFluidIngredient.java

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.fluids.crafting;
+
+import com.mojang.datafixers.util.Either;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.DataResult;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import io.netty.buffer.ByteBuf;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import net.minecraft.network.RegistryFriendlyByteBuf;
+import net.minecraft.network.codec.ByteBufCodecs;
+import net.minecraft.network.codec.StreamCodec;
+import net.minecraft.tags.TagKey;
+import net.minecraft.world.level.material.Fluid;
+import net.neoforged.neoforge.common.util.NeoForgeExtraCodecs;
+import net.neoforged.neoforge.fluids.FluidStack;
+import net.neoforged.neoforge.fluids.FluidUnit;
+import net.neoforged.neoforge.fluids.FluidUnitSpec;
+import net.neoforged.neoforge.fluids.FluidUnits;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Standard implementation for a FluidIngredient with an amount.
+ *
+ * <p>{@link FluidIngredient}, like its item counterpart, explicitly does not perform count checks,
+ * so this class is used to (a) wrap a standard FluidIngredient with an amount and (b) provide a
+ * standard serialization format for mods to use.
+ *
+ * @see net.neoforged.neoforge.common.crafting.SizedIngredient
+ */
+public final class UnitAmountFluidIngredient {
```

Comment:
Making this non-final may be useful to support other matching methods (e.g. exact match) without reimplementing the logic.

Alternatively, make some static helpers for anything complex so someone can make their own unit amount fluid ingredient.

Reply:
Moved the logic to a static function